crates/server/src/datastore/mod.rs

```diff
@@ -111,7 +116,16 @@ pub trait DataStore: Send + Sync {
     /// This is an expensive operation and should only be performed on startup.
     async fn get_all_validated_records(
         &self,
-    ) -> Result<Pin<Box<dyn Stream<Item = Result<LogLeaf, DataStoreError>> + Send>>, DataStoreError>;
+    ) -> Result<
+        Pin<Box<dyn Stream<Item = Result<(RegistryIndex, LogLeaf), DataStoreError>> + Send>>,
```

Comment:
Isn't the `RegistryIndex` guaranteed to always be `0..N`?

Reply:
I think so? You can always get the index later with e.g. https://docs.rs/futures/0.3.28/futures/stream/trait.StreamExt.html#method.enumerate if you need it.

Reply:
...though if I'm reading the code correctly I think `State` could just fill in the index itself.

Reply:
Reworking and pushing shortly. Agree and see how to simplify.

Reply:
Just pushed